components/WaveHandPicker.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { useState } from 'react';
+import { WaveHandButton } from '@/components/WaveHandButton';
+import {HandWaveBadge } from '@/lib/draw_badge_api';
+
+const waveHandEmoji = '👋';
+
+interface WaveHandPickerProps {
+  initialHands: string[];
+  updateHandWaveBadge: (badge: HandWaveBadge) => void;
+}
+
+export function WaveHandPicker({
+  initialHands,
+  updateHandWaveBadge,
+}: WaveHandPickerProps) {
+
+  const [selectedButtonId, setSelectedButtonId] = useState(-1);
+
+  const appendEmoji = (text: string) => {
+    return waveHandEmoji + ' ' + text;
+  }
+
+  const clickButton = (id: number, text: string) => {
+    if (selectedButtonId == id) {
+      setSelectedButtonId(-1); // unselect a button
+      // hide HandWaveBadge
+      updateHandWaveBadge({visible: false});
+    } else { 
+      setSelectedButtonId(id);
+      // draw selected HandWaveBadge
+      updateHandWaveBadge({
+        visible: true,
+        waveText: appendEmoji(text),
+      })
+    }
+  };
+
+  return (
+    <div className="button-rows">
+      {initialHands.map((text, index) => (
+        <WaveHandButton
+          key={'wave-hand-button-' + index}
```

Comment:
Note, on https://react.dev/learn/rendering-lists, it cautions against using the array index to compute the key:

>  You might be tempted to use an item’s index in the array as its key. In fact, that’s what React will use if you don’t specify a key at all. But the order in which you render items will change over time if an item is inserted, deleted, or if the array gets reordered. Index as a key often leads to subtle and confusing bugs.

For this reason, I think that using `text` rather than `index` would be preferable.

Reply:
Replaced with `text`.